samtranslator/model/eventsources/push.py

```diff
@@ -407,13 +407,16 @@ def _inject_notification_configuration(self, function, bucket):
             notification_config = {}
             properties["NotificationConfiguration"] = notification_config
 
+        if not isinstance(notification_config, dict):
+            raise InvalidResourceException(bucket_id, "Invalid type for NotificationConfiguration. Must be a dict.")
```

Comment:
Is `dict` the right wording here? I only ask because `dict` is pretty Python specific. How does CloudFormation communicate things like this?

Reply:
CloudFormation doesn't seem have a generic name for an object/dict (e.g. it's called [`NotificationConfiguration`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-notificationconfig.html) type for this case).

Instead of "dict", I think we can just say "Invalid type for NotificationConfiguration". I think it's informative enough for customer to find out how to fix it.